redis/tests/test_streams.rs

```diff
@@ -1955,3 +1940,125 @@ fn test_xrevrange() {
+    // Note that this issue also applies to xclaim.
+
+    let ctx = TestContext::new();
+    let mut con = ctx.connection();
+
+    // xautoclaim-invalid basic idea:
+    // 1. add messages to a group
+    // 2. read the messages, but do not xack them
+    // 3. delete the messages from the stream
+    // 4. call xautoclaim
+
+    let reply = create_group_add_and_read(&mut con);
+
+    // save this StreamId for later
+    let claim = &reply.keys[0].ids[0];
+    let claim_1 = &reply.keys[0].ids[1];
+    let claimed_entries = vec![reply.keys[0].ids[2].clone(), reply.keys[0].ids[3].clone()];
+
+    // delete the messages from the stream
+    let _ = con.xdel("k1", &[claim.id.clone(), claim_1.id.clone()]);
+    sleep(Duration::from_millis(5));
+
+    let reply = con
+        .xautoclaim_options(
+            "k1",
+            "g1",
+            "c2",
+            1,
+            claim.id.clone(),
+            StreamAutoClaimOptions::default().count(4),
+        )
+        .unwrap();
+    assert_eq!(reply.claimed, claimed_entries);
+
+    if ctx.get_version().0 > 6 {
+        assert_eq!(
+            reply.deleted_ids,
+            vec![claim.id.clone(), claim_1.id.clone()]
+        );
+    } else {
+        assert_eq!(reply.deleted_ids.len(), 0);
+        assert!(reply.invalid_entries);
+    }
+}
+
+#[test]
+fn test_xautoclaim_invalid_pel_entries_claiming_just_ids() {
+    // The Redis PEL can include stale entries that have been deleted from the stream,
+    // due to either data corruption or client error.
+    // Redis v6 behaves differently from Redis v7 when encountering these invalid entries.
+    // We support v6, so we must ensure that stale entries do not cause a deserialization error.
+    // See https://github.com/redis-rs/redis-rs/issues/1798
+    // Note that this issue also applies to xclaim.
+
+    let ctx = TestContext::new();
+    let mut con = ctx.connection();
+
+    // xautoclaim-invalid basic idea:
+    // 1. add messages to a group
+    // 2. read the messages, but do not xack them
+    // 3. delete the messages from the stream
+    // 4. call xautoclaim
+
+    let reply = create_group_add_and_read(&mut con);
+
+    // save this StreamId for later
+    let claim = &reply.keys[0].ids[0];
+    let claim_1 = &reply.keys[0].ids[1];
+    let mut claimed_entries = vec![reply.keys[0].ids[2].clone(), reply.keys[0].ids[3].clone()];
+
+    // delete the messages from the stream
+    let _ = con.xdel("k1", &[claim.id.clone(), claim_1.id.clone()]);
+    sleep(Duration::from_millis(5));
+
+    let reply = con
+        .xautoclaim_options(
+            "k1",
+            "g1",
+            "c2",
+            1,
+            claim.id.clone(),
+            StreamAutoClaimOptions::default().count(4).with_justid(),
+        )
+        .unwrap();
+    // we expect to receive just IDs, so we remove the maps
+    std::mem::take(&mut claimed_entries[0].map);
+    std::mem::take(&mut claimed_entries[1].map);
+
+    if ctx.get_version().0 > 6 {
```

Comment:
I didn't take a close look at how the test context works; is the context using `REDIS_RS_REDIS_JSON_PATH`? 
I wasn't able to get that working for Redis 6 the other day.

Reply:
It uses `REDIS_RS_REDIS_JSON_PATH` only when the JSON module is specified. When no module is specified you don't need to build the JSON module.

Reply:
So what redis engine is used by the tests?

Reply:
The [CI matrix is running](https://github.com/redis-rs/redis-rs/blob/3d2a103f89d59256fd76c0ccb38779692a1412a6/.github/workflows/rust.yml#L24) Redis 6.2, 7.2, 8.0, 8.2 & valkey 7.2 and 8.0.